autorest/codegen/models/operation.py

```diff
@@ -107,15 +107,17 @@ def has_request_body(self):
 
     @property
     def body_parameter(self) -> Parameter:
-        if not self.has_request_body:
-            raise ValueError(f"There is no body parameter for operation {self.name}")
+        if not self.has_request_body or not self.parameters:
```

Comment:
I want to keep the original test, we can add in the `__init__`
```python
if parameters is None:
    parameters = []
```
that should do the trick